esp/services/ws_workunits/ws_wudetails.hpp

```diff
@@ -0,0 +1,126 @@
+        unsigned __int64 getExactValue() { return exactValue; }
+        unsigned __int64 getMinValue()   { return minValue; }
+        unsigned __int64 getMaxValue()   { return maxValue; }
+        bool hasExactValue()             { return bExactValue; }
+        bool hasMinValue()               { return bMinValue; }
+        bool hasMaxValue()               { return bMaxValue; }
+    };
+
+    static int funcCompareAttribFilter(IInterface* const * l, IInterface* const * r)
+    {
+        CAttributeFilter * ll = (dynamic_cast<CAttributeFilter *>(*l));
+        CAttributeFilter * rr = (dynamic_cast<CAttributeFilter *>(*r));
+
+        if (ll->getStatisticKind()>rr->getStatisticKind()) return +1;
+        if (ll->getStatisticKind()<rr->getStatisticKind()) return -1;
+        return 0;
+    }
+    static int funcCompareUnsignedShort(const unsigned short * l, const unsigned short * r)
+    {
+        if (*l>*r) return +1;
+        if (*l<*r) return -1;
+        return 0;
+    }
+    static int funcCompareString(char const * const *l, char const * const *r)
+    {
+      return strcmp(*l, *r);
+    }
+    StatisticsFilter * initStatisticsFilter(IConstWUScopeFilter & filter,
+                                            IConstWUAttributeToReturn & attribsToReturn);
+
+    void buildAttribFilter(IArrayOf<CAttributeFilter> & processedAttribFilter,
+                           IArrayOf<IConstWUAttributeFilter> & reqAttribFilter);
+
+    void buildAttribListToReturn(UnsignedShortArray & skReturnAttribList,
+                                 StringArray & returnAttriblist);
+
+    void buildScopeTypeList(UnsignedShortArray & processedScopeTypeList,
+                            StringArray & arrayScopeTypeList);
+
+    void processWUStatsIntoResponse(IArrayOf<IEspWUScope> & respScopes,
+                                    IConstWUStatisticIterator & statsIter,
+                                    UnsignedShortArray & nestedScopeTypes,
+                                    UnsignedShortArray & returnAttribList,
+                                    StringArray & ids,
+                                    IArrayOf<CAttributeFilter> & arrayAttributeFilter,
+                                    unsigned __int64 minVersion,
+                                    unsigned __int64 & maxTimestamp);
+
+    IEspWUScope *buildRespWUScope(SCMStringBuffer & scope,
+                                  SCMStringBuffer & scopeType,
+                                  SCMStringBuffer & id,
+                                  IArrayOf<IEspWUResponseAttribute> & attribs);
+
+    IEspWUResponseAttribute *copyAttribs(IConstWUStatistic & cur,
+                                         IConstWUAttributeOptions & attribOptions);
+
+    SCMStringBuffer & getScopeIdFromScopeName(SCMStringBuffer & scopeId,
+                                              SCMStringBuffer & scopeName);
+    inline unsigned getScopeDepth(SCMStringBuffer & sScope)
```

Comment:
No reason this should be in the header or a member function.  Another example where a const char * parameter would be better.